aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -801,53 +802,60 @@ C10_LAUNCH_BOUNDS_1(512)
 #endif
 __global__
 void triu_tril_kernel(
-    scalar_t* result, scalar_t* self, int64_t k, int64_t N,
-    int64_t res_batch_stride, int64_t res_row_stride, int64_t res_col_stride,
-    int64_t self_batch_stride, int64_t self_row_stride, int64_t self_col_stride, int64_t self_ncol) {
+    cuda::detail::TensorInfo<scalar_t, int64_t> result_info,
```

Comment:
Also, if performance is important, then templating on index type and using int32_t for smaller tensors helps a lot.

Reply:
Thank you for the comment. In this case, I would have to check if 32 bit indexing can be used, and then use the appropriate index type, correct?